### PR TITLE
Add bonuses for crew achievements

### LINKED ISSUE #712: Add bonuses for crew achievements

### PR DESCRIPTION
**Pull request by nyet-ty**
Resolve #712

## PR includes

-   [x] Feature

### CODE REVIEW

**Review by 9teen90nine — APPROVED** [review]

File: prisma/schema.prisma [hunk 1]
```diff
@@ -244,6 +244,18 @@ model BonusRule {
   crewAchievement            Achievement[]
 }
 
+model BonusForAchievementRule {
+  id                        String      @id @default(nanoid())
+  bonusesPerCrewAchievement Int
+  description               String
+  crewAchievementId         String
```
Comment (tsumo) [inline prisma/schema.prisma:251]: Maybe switching names would be better

achievementId - for crew achievement

externalAchievementId - for external achievement data

File: src/modules/achievementMethods.ts [hunk 2]
```diff
@@ -50,6 +60,26 @@ export const achievementMethods = {
             return user;
         }
 
+        if (achievement.bonusForAchievementRule && config.techAdminId) {
```
Comment (tsumo) [inline src/modules/achievementMethods.ts:63]: What is techAdminId?

**Review by tsumo — CHANGES_REQUESTED** [review]

File: src/modules/achievementMethods.ts (hunk 2 above)
Comment (nyet-ty) [reply src/modules/achievementMethods.ts:63]: technical account to give bonuses and achievements automatically

File: prisma/schema.prisma (hunk 1 above)
Comment (nyet-ty) [reply prisma/schema.prisma:251]: sounds right, thanks

File: prisma/migrations/20240426181554_bonus_for_achievement/migration.sql [hunk 3]
```diff
@@ -0,0 +1,30 @@
+/*
+  Warnings:
+
+  - You are about to drop the column `achievementCategory` on the `BonusHistory` table. All the data in the column will be lost.
+  - You are about to drop the column `achievementId` on the `BonusHistory` table. All the data in the column will be lost.
+
+*/
+-- AlterTable
+ALTER TABLE "BonusHistory" DROP COLUMN "achievementCategory",
+DROP COLUMN "achievementId",
```
Comment (tsumo) [inline prisma/migrations/20240426181554_bonus_for_achievement/migration.sql:10]: Drop should be replaced with rename to save the data

**Review by tsumo — CHANGES_REQUESTED** [review]

**Review by tsumo — APPROVED** [review]

**Pull request merged**